signer/src/testing/network.rs

```diff
@@ -45,9 +45,17 @@ fn spawn_client_task(
             client.broadcast(msg).await.expect("Failed to broadcast");
         }
 
-        for msg in should_receive {
+        for _ in 0..should_receive.len() {
             let received = client.receive().await.expect("Failed to receive message");
-            assert_eq!(received, msg);
+            assert!(should_receive.contains(&received));
         }
+
+        // !! This is the original implementation, but it assumes exact ordering of
+        // !! messages, which is not guaranteed in the network (this makes tests
+        // !! using libp2p's gossipsub flakey).
+        // for msg in should_receive {
+        //     let received = client.receive().await.expect("Failed to receive message");
+        //     assert_eq!(received, msg);
+        // }
```

Comment:
Very useful comment. I would delete the commented out code here and move the comment about how we cannot assume exact ordering of messages to the modified code above.

Reply:
Yeah the comment was more for you guys (reviewers) to understand why I changed it. Updated in [7173ec1](https://github.com/stacks-network/sbtc/pull/473/commits/7173ec1c34c181344ea16407bb81ccf4e4c260ad)